app/assets/javascripts/Components/Result/file_viewer.jsx

```diff
@@ -39,15 +39,24 @@ export class FileViewer extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (!this.props.result_id && prevProps.selectedFile !== this.props.selectedFile) {
+    let fileManagerChanged = !this.props.result_id && prevProps.selectedFile !== this.props.selectedFile;
+    let feedbackFileChanged = !!this.props.feedbackFile && prevProps.feedbackFile !== this.props.feedbackFile
+    if (fileManagerChanged || feedbackFileChanged) {
       this.set_submission_file(null);
     }
   }
 
   setFileUrl = (submission_file_id) => {
     let url;
     if (!this.props.result_id) {
-      url = this.props.selectedFileURL
+      url = this.props.selectedFileURL;
+    } else if (!!this.props.feedbackFile) {
+      url = Routes.get_feedback_file_assignment_submission_path(
```

Comment:
@david-yz-liu  
Currently, this is how images are sourced as feedback files. However, the route that handles this sends images differently than the download route (see [this line](https://github.com/MarkUsProject/Markus/pull/4813/files#diff-2d02aa9ac63d85a26427a962afd38387L67) in the old code of feedback file panel). Overall, the testing for this PR should be more thorough than what I had a chance to do.